cli/src/commands/workspace/list.rs

```diff
@@ -17,26 +17,60 @@ use tracing::instrument;
 
 use crate::cli_util::CommandHelper;
 use crate::command_error::CommandError;
+use crate::commit_templater::WorkspaceRef;
 use crate::ui::Ui;
 
 /// List workspaces
 #[derive(clap::Args, Clone, Debug)]
-pub struct WorkspaceListArgs {}
+pub struct WorkspaceListArgs {
+    /// Render each workspace using the given template
+    ///
+    /// All 0-argument methods of the [`WorkspaceRef` type] are available as
+    /// keyword in the template expression. See [`jj help -k templates`] for
+    /// more information.
+    ///
+    /// [`WorkspaceRef` type]:
+    ///     https://jj-vcs.github.io/jj/latest/templates/#workspaceref-type
+    ///
+    /// [`jj help -k templates`]:
+    ///     https://jj-vcs.github.io/jj/latest/templates/
+    #[arg(long, short = 'T')]
+    template: Option<String>,
+}
 
 #[instrument(skip_all)]
 pub fn cmd_workspace_list(
     ui: &mut Ui,
     command: &CommandHelper,
-    _args: &WorkspaceListArgs,
+    args: &WorkspaceListArgs,
 ) -> Result<(), CommandError> {
     let workspace_command = command.workspace_helper(ui)?;
     let repo = workspace_command.repo();
     let mut formatter = ui.stdout_formatter();
-    let template = workspace_command.commit_summary_template();
+    let commit_template = workspace_command.commit_summary_template();
+
+    let maybe_template = args
+        .template
+        .as_ref()
+        .map(|text| {
+            let language = workspace_command.commit_template_language();
+
+            workspace_command.parse_template(ui, &language, text)
+        })
+        .transpose()?;
+
     for (name, wc_commit_id) in repo.view().wc_commit_ids() {
-        write!(formatter, "{}: ", name.as_symbol())?;
         let commit = repo.store().get_commit(wc_commit_id)?;
-        template.format(&commit, formatter.as_mut())?;
+
+        if let Some(template) = &maybe_template {
+            let ws_ref = WorkspaceRef::new(name.clone(), commit);
+            template.format(&ws_ref, formatter.as_mut())?;
+        } else {
+            // Fallback to default formatting
+            write!(formatter, "{}: ", name.as_symbol())?;
+            commit_template.format(&commit, formatter.as_mut())?;
+        }
```

Comment:
Can you add the default `templates.workspace_list` which will print `name` and `format_commit_summary_with_refs()`?

Reply:
It looks like `format_commit_summary_with_refs` takes `refs` too, where do I get them?

Reply:
Just copy what `commit_summary` will print:
`format_commit_summary_with_refs(target, target.bookmarks()`

If the user override the `commit_summary`, they'll have to override `workspace_list` template as well, but that's basically the same as the other default templates.